clients/src/main/java/org/apache/kafka/common/header/RecordHeader.java

```diff
@@ -14,33 +14,27 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.apache.kafka.common.record;
+package org.apache.kafka.common.header;
 
-import org.apache.kafka.common.utils.Utils;
-
-import java.nio.ByteBuffer;
+import java.util.Arrays;
 import java.util.Objects;
 
-public class Header {
+public class RecordHeader implements Header {
     private final String key;
-    private final ByteBuffer value;
+    private final byte[] value;
```

Comment:
Yes it was, as the KIP interface and constructor for a header is byte[]. When we create the headers in producer it will be a byte[] as such this would not be any memory copy, like wise on consume when the header is read a byte[] would need to be returned as such any saving would be negated.

If we want ByteBuffer, then it would be best to change that you construct Headers with a ByteBuffer value, and like wise byte[] value(),  changes to ByteBuffer value(). 

Im not opposed to this, just i'snt as agreed in KIP.